src/ConnectedFieldArray.js

```diff
@@ -37,8 +37,11 @@ const createConnectedFieldArray = (structure: Structure<*, *>) => {
       const nextValue = nextProps.value
 
       if (thisValue && nextValue) {
+          let nextValueItemsSame = nextValue.every(val => thisValue.includes(val));
+          let nextValueItemsOrderChanged = nextValue.some((val, index) => val !== thisValue[index]);
         if (
           thisValue.length !== nextValue.length ||
+          (nextValueItemsSame && nextValueItemsOrderChanged) ||
```

Comment:
@erikras Is there a reason this is an `&&`? If the `nextValueItemsSame` is `false`, shouldn't we always rerender? In this case, it'll only rerender if the items have changed AND they've rearranged.

In our use-case, we're deleting an item from a field list array. We add `destroy: true` to the value object and want to rerender the list with the "destroyed" item hidden. The list does not rerender because of this logic.